rudder-core/src/main/scala/com/normation/rudder/datasources/DataSource.scala

```diff
@@ -39,11 +39,21 @@ package com.normation.rudder.datasources
 
 import org.joda.time.DateTime
 import org.joda.time.Seconds
+import net.liftweb.common._
 
 sealed trait DataSourceType {
   def name : String
 }
 
+object DataSourceType {
+  def apply(name : String) : Box[DataSourceType] = {
+    if (name == ByNodeSourceType.name) {
```

Comment:
Not sure I understand why only ByNode?
The generic way for that is to add a "val all = ca.mrvisser.sealerate.values[DataSourceType]"
And then use a "all.find(name.toLowerCase == _.name.toLowercase) match {case None => Failure etc"

To make all work, you also need to put you case object into DataSourceType.
And they should be mark "final", because scala is stupid.